src/logging/OTELLogger.ts

```diff
@@ -0,0 +1,291 @@
+/*
+Copyright 2023 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { trace, propagation, Tracer, Span, Context, ROOT_CONTEXT, AttributeValue, Attributes, SpanKind } from "@opentelemetry/api";
+import { SemanticAttributes } from "@opentelemetry/semantic-conventions";
+
+import { LogFilter, LogLevel } from "./LogFilter";
+import { FilterCreator, ILogger, ILogItem, ILogReporter, ISerializedItem, LabelOrValues, LogCallback, LogItemValues } from "./types";
+import type { Clock } from "../platform/web/dom/Clock.js";
+
+function itemCaption(values: LogItemValues): string {
+    if (values.t === "network") {
+        return `HTTP request`;
+    } else {
+        return values.l || values.t || "Log item";
+    }
+}
+
+function mapLabelToAttribute(t: string | undefined, key: string): string | null {
+    if (key === "l") return null;
+    if (key === "t") return null;
+
+    if (t === "network") {
+        if (key === "method") return SemanticAttributes.HTTP_METHOD;
+        if (key === "url") return SemanticAttributes.HTTP_URL;
+        if (key === "status") return SemanticAttributes.HTTP_STATUS_CODE;
+    }
+
+    return `hydrogen.${key}`;
+}
+
+function labelsToAttributes(labels: object): Attributes {
+    return Object.entries(labels)
+        .reduce((attrs, [key, value]) => {
+            const mapped = mapLabelToAttribute(labels['t'], key);
+            if (!mapped) return attrs;
+            return { ...attrs, [mapped]: value as AttributeValue };
+        }, {
+        });
+}
```

Comment:
Those functions are an attempt to map the various keys in the log entries to well-known OTEL attributes. The list of those attributes can be found here: https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/README.md